docs/roman/datamodels/models.rst

```diff
@@ -8,58 +8,84 @@ the underlying file format.  The same data model may be used for data
 
-Hierarchy of models
--------------------
-
-There are different data model classes for different kinds of data.
-
-One model instance, many arrays
--------------------------------
-
-Each model instance generally has many arrays that are associated with
-it.  For example, the `ImageModel` class has the following arrays
-associated with it:
+Each model instance is created to contain a variety of attributes and data that
+are for needed for analysis or to propogate information about the file and the
+contents of the file. For example, the `ImageModel` class has the following
+arrays associated with it:
 
     - `data`: The science data
     - `dq`: The data quality array
     - `err`: The error array
 
-The shape of these arrays must be broadcast-compatible.  If you try to
-assign an array to one of these members that is not
-broadcast-compatible with the data array, an exception is raised.
+The shape of these arrays must be broadcast-compatible. If you try to
+assign an array to one of these members that is not broadcast-compatible with
+the data array, an exception is raised.
+
+Along with data arrays the datamodel also contains information about the
+observation that can include the Observation program, exposure information,
+pointing informtation and processing steps.
 
 Working with models
 ===================
 
+Reading a data model
+--------------------
+
+If you have an exisiting data file it is straightforward to access the file
+using python.
+
+.. code-block:: python
+
+    from roman_datamodels import datamodels as rdm
+    fn = 'r0019106003005004023_03203_0034_WFI01_cal.asdf'
+    data_file = rdm.open(fn)
+    type(data_file)
+
+Where the output of the type command tells you that you have imported an
+ImageModel from roman_datamodels,
+
+
+    <class 'roman_datamodels.datamodels.ImageModel'>
+
 Creating a data model from scratch
 ----------------------------------
 
 To create a new `ImageModel`, just call its constructor.  To create a
 new model where all of the arrays will have default values, simply
 provide a shape as the first argument::
 
-.. doctest-skip::
+    from roman_datamodels import datamodels as rdm
+    from roman_datamodels.testing.factories import create_wfi_image
+    import numpy as np
 
-    from romancal.datamodels import ImageModel
-    with ImageModel((1024, 1024)) as im:
+    image_node = create_wfi_image()
+    image_model = rdm.ImageModel(image_node)
+    type(image_model)
+    #<class 'roman_datamodels.datamodels.ImageModel'>
+
+.. note ::
+    The values in the file generated by create_wfi_image are intended to be
+    clearly incorrect and should be replaced if the file is intended to be used
+    for anything besides a demonstration.
 
-        ...
 
 In this form, the memory for the arrays will not be allocated until
 the arrays are accessed.  This is useful if, for example, you don’t
 need a data quality array -- the memory for such an array will not be
 consumed::
```

Comment:
I don't believe this is true for roman_datamodels.  The memory is allocated by create_wfi_image.